clang/lib/AST/TypePrinter.cpp

```diff
@@ -1635,6 +1635,14 @@ void TypePrinter::printElaboratedBefore(const ElaboratedType *T,
     if (T->getKeyword() != ElaboratedTypeKeyword::None)
       OS << " ";
     NestedNameSpecifier *Qualifier = T->getQualifier();
+    if (Policy.FullyQualifiedName) {
```

Comment:
`FullyQualifiedName` controls printing of identifier qualification, not elaboration of type names. IOW, it controls printing `Bar` vs `Foo::Bar` rather than `B` vs `struct B`.

I think you want to be looking at `!SuppressTagKeyword` right?